### PR TITLE
feat: add Batch aspect

### PR DESCRIPTION
**Pull request by amliuyong**
Fixes # add aspect for batch job

### CODE REVIEW

**Comment by zxkane** [comment]
LGTM

**Review by zxkane — APPROVED** [review]

File: src/aspect.ts [hunk 1]
```diff
@@ -14,7 +16,7 @@ export interface ECRRepositoryAspectProps {
    *
    * @default - process.env.BSS_IMAGE_ASSET_ACCOUNT_ID
    */
-  readonly imageAssetAccountId?: string;
+  readonly imageAssetAccountId ? : string;
```
Comment (wchaws) [inline src/aspect.ts:19]: I prefer not to add a space in between.
please use `readonly imageAssetAccountId ? : string;`

File: src/aspect.ts [hunk 2]
```diff
@@ -137,11 +151,74 @@ export class StepFunctionsSageMakerTrainingJob extends ECRRepositoryAspect {
...
+ */
+
+export class BatchJobDefinitionAspect extends ECRRepositoryAspect {
+  /**
+   * @internal
+   */
+  readonly _repoNames: string[];
+  private _executionRole ? : iam.Role;
+  private _executionRoleArn ? : string;
+  private _allRolesMap: Map < string, iam.Role >;
```
Comment (wchaws) [inline src/aspect.ts:166]: I prefer not to add a space in between.

File: src/aspect.ts [hunk 3]
```diff
@@ -27,7 +29,7 @@ export abstract class ECRRepositoryAspect implements IAspect {
   /**
    * @internal
    */
-  static readonly _repoPolicies = new Map<string, Policy>();
+  static readonly _repoPolicies = new Map < string, Policy > ();
```
Comment (wchaws) [inline src/aspect.ts:32]: I prefer not to add a space in between. i.e. keep `static readonly _repoPolicies = new Map<string, Policy>();`

**Review by wchaws — CHANGES_REQUESTED** [review]

File: src/aspect.ts (hunk 1 above)
Comment (zxkane) [reply src/aspect.ts:19]: Looks like it's changed by `eslint`, something is introduced in recent update?

Comment (wchaws) [reply src/aspect.ts:19]: Let me check

Comment (zxkane) [reply src/aspect.ts:19]: @wchaws any finding?

Comment (wchaws) [reply src/aspect.ts:19]: @zxkane I've checked in gitpod env. eslint won't add space.
And eslint allow this kind of syntax. I guest it's caused by editor. 
Again, I still prefer to manually eliminate those spaces.

**Review by wchaws — APPROVED** [review]
LGTM

**Pull request merged**